From 6497c54d598bd7b62c2a54a76d3246b155a1cc4b Mon Sep 17 00:00:00 2001
From: Seung-Woo Kim <sw0312.kim@samsung.com>
Date: Thu, 4 Dec 2014 20:19:30 +0900
Subject: [PATCH 29/34] input: keyboard: mcs_touchkey: add sentinel into array
 of of_device_id

Without sentinel, of_match_node() to array causes out-of-bound
memory access. So this patch adds sentinel into array.

Change-Id: Icfe3d8c2fdb0faa006f413e7c080b6d713eab08c
Signed-off-by: Seung-Woo Kim <sw0312.kim@samsung.com>
---
 drivers/input/keyboard/mcs_touchkey.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/drivers/input/keyboard/mcs_touchkey.c b/drivers/input/keyboard/mcs_touchkey.c
index 933429c165b5..21ceea6bbff5 100644
--- a/drivers/input/keyboard/mcs_touchkey.c
+++ b/drivers/input/keyboard/mcs_touchkey.c
@@ -321,6 +321,7 @@ MODULE_DEVICE_TABLE(i2c, mcs_touchkey_id);
 static struct of_device_id mcs_touchkey_dt_match[] = {
 	{ .compatible = "mcs5000_touchkey", },
 	{ .compatible = "mcs5080_touchkey", },
+	{ },
 };
 
 static struct i2c_driver mcs_touchkey_driver = {
-- 
2.20.1

